Plugins/GenerateDoccReference/GenerateDoccReference.swift

```diff
@@ -18,83 +21,101 @@ struct GenerateDoccReferencePlugin: CommandPlugin {
+    let buildArtifacts = try buildPackage(configuration: configuration)
+
+    try generateDocs(context: context, with: extractor, buildArtifacts: buildArtifacts, generationToolFile: generationToolFile)
+  }
 
-    // Build all products first.
-    print("Building package in \(configuration) mode...")
+  private func buildPackage(configuration: PackageManager.BuildConfiguration) throws -> [PackageManager.BuildResult.BuiltArtifact] {
+    // The package must be built before documentation can be generated
+    // because DocC requires compiled executables to extract symbols.
     let buildResult = try packageManager.build(
       .all(includingTests: false),
       parameters: .init(configuration: configuration))
 
+    // If the build fails, documentation generation cannot proceed.
     guard buildResult.succeeded else {
       throw GenerateDoccReferencePluginError.buildFailed(buildResult.logText)
     }
+
     print("Built package in \(configuration) mode")
+    return buildResult.builtArtifacts
+  }
 
-    // Run generate-docc-reference on all executable artifacts.
-    for builtArtifact in buildResult.builtArtifacts {
-      // Skip non-executable targets
+  private func generateDocs(
+    context: PluginContext, with extractor: ArgumentExtractor,
+    buildArtifacts: [PackageManager.BuildResult.BuiltArtifact],
+    generationToolFile: Path
+  ) throws {
+    for builtArtifact in buildArtifacts {
+      // DocC documentation is only relevant for executable targets,
+      // since documentation is typically generated for command-line tools.
       guard builtArtifact.kind == .executable else { continue }
 
-      // Skip executables without a matching product.
+      // Ensure the executable is associated with a product, since some
+      // build artifacts may not map directly to user-facing products.
       guard let product = builtArtifact.matchingProduct(context: context)
       else { continue }
 
-      // Skip products without a dependency on ArgumentParser.
+      // Skip products that do not depend on ArgumentParser because
+      // this plugin is specifically designed to generate documentation
+      // for command-line tools built using ArgumentParser.
       guard product.hasDependency(named: "ArgumentParser") else { continue }
 
-      // Skip products with multiple underlying targets.
+      // If a product has multiple targets, it's unclear which should
+      // be documented. Restricting to single-target products avoids ambiguity.
       guard product.targets.count == 1 else { continue }
       let target = product.targets[0]
 
-      // Get the artifacts name.
       let executableName = builtArtifact.path.lastComponent
       print("Generating docc reference for \(executableName)...")
 
-      // Create output directory.
+      // Organizing documentation by target name ensures that multiple
+      // command-line tools within the same package don't overwrite each
+      // other's generated documentation.
```

Comment:
I appreciate the focus on "why" rather than "what" but this is not correct.

A documentation catalog (.docc directory) is associated with a specific target based on the catalog existing in the target's sources directory. Each target only has a single documentation catalog so the `target.name` part of this code doesn't matter. What matter is that the output directory is in the target's sources directory. 

To me, this documentation makes it sound like the output directory path need to include the target name to function correctly but it's not needed and it could be the wrong output path. If the target already has a documentation catalog, this code might create a second catalog (which won't work).